packages/plugin-js-packages/src/lib/runner/outdated/transform.ts

```diff
@@ -18,23 +14,27 @@ export function outdatedResultToAuditOutput(
   const relevantDependencies: OutdatedResult = result.filter(
     dep => dep.type === dependencyGroupToLong[depGroup],
   );
-  // TODO use semver logic to compare versions
-  const outdatedDependencies = relevantDependencies
-    .filter(dep => dep.current !== dep.latest)
-    .filter(
-      dep =>
-        dep.current.split('-')[0]?.toString() !==
-        dep.latest.split('-')[0]?.toString(),
-    );
 
-  const outdatedStats = outdatedDependencies.reduce(
-    (acc, dep) => {
-      const outdatedLevel = getOutdatedLevel(dep.current, dep.latest);
-      return { ...acc, [outdatedLevel]: acc[outdatedLevel] + 1 };
-    },
-    { major: 0, minor: 0, patch: 0 },
+  const validDependencies = relevantDependencies
+    .filter(dep => parse(dep.current) != null && parse(dep.latest) != null)
+    .map(dep => ({
+      ...dep,
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      current: clean(dep.current)!,
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      latest: clean(dep.latest)!,
+    }));
```

Comment:
> Can I get rid of the non-null-assertion? I use semver functions to make sure to filter out null values and up-to-date dependencies but TS does not know.

You can use a type predicate if you swap the order - first call `clean()` in `.map()`, then check both are non-null (i.e. valid) in `.filter()`. According to the docs, both `parse()` and `clean()` return `null` for invalid versions, so don't need to call both.

```suggestion
  const validDependencies = relevantDependencies
    .map(dep => ({
      ...dep,
      current: clean(dep.current),
      latest: clean(dep.latest),
    }))
    .filter(
      (dep): dep is OutdatedResult[number] =>
        dep.current != null && dep.latest != null,
    );
```

Reply:
Nice, thank you